### DISCUSSION TITLE
AccessibleIcon and VisuallyHidden can't use in SSR

### DISCUSSION THREAD

**Discussion by davidkhierl** [radix-ui/primitives#2482] [Help]
I see `AccessibleIcon` uses `VisuallyHidden`, I'm using nextjs but whenever I uses this components I need to put `"use client"` directives. I checked the source code and it only uses `cloneElement`, I'm not sure if this one the one that causing the error because the error says the component uses `useEffect`

**Comment by benoitgrelard** [comment]
I think it's because of the underlying `Primitive` which has an effect to add a symbol to the window to know when Radix is used on a page. We should probably remove this once we add proper RSC support.
